autoload/lsp.vim

```diff
@@ -880,45 +913,35 @@ function! lsp#get_allowed_servers(...) abort
         endif
     endif
 
-    " TODO: cache active servers per buffer
-    let l:active_servers = []
+    if has_key(s:active_servers, l:buffer_filetype)
+      let l:cached_servers = s:active_servers[l:buffer_filetype]
+      if get(l:cached_servers, 'update_cnt', -1) == s:servers_update_cnt
+        " copy the list to prevent inplace changes
+        return copy(l:cached_servers['servers'])
+      endif
+    endif
 
-    for l:server_name in keys(s:servers)
-        let l:server_info = s:servers[l:server_name]['server_info']
-        let l:blocked = 0
+    " TODO: cache active servers per buffer
+    let l:update_cnt = s:servers_update_cnt
 
-        if has_key(l:server_info, 'blocklist')
-            let l:blocklistkey = 'blocklist'
-        else
-            let l:blocklistkey = 'blacklist'
-        endif
-        if has_key(l:server_info, l:blocklistkey)
-            for l:filetype in l:server_info[l:blocklistkey]
-                if l:filetype ==? l:buffer_filetype || l:filetype ==# '*'
-                    let l:blocked = 1
-                    break
-                endif
-            endfor
-        endif
+    if g:lsp_use_lua && lsp#utils#has_lua()
+      let l:active_servers = empty(s:servers) ? [] : luaeval(
+            \ 'require("lsp/lsp").get_allowed_servers(_A.bf, _A.s)',
```

Comment:
If the main module is under `lsp/init.lua` then you can just use `require 'lsp'` which would read better. `init.lua` is treated special by both vim and neovim.